IFT-Explainer.md

```diff
@@ -0,0 +1,174 @@
+complex inter-relationships between different OpenType™ tables,
+or when characters are **shared between multiple writing systems** but
+behave **differently** in each one.
+
+Current subsetting does not address subsetting **design axes** (variable fonts).
+
+## Non-goals
+
+Changes to the Open Font Format or OpenType specifications are out of scope.
+
+## Considered alternatives
+
+A 2020 [Evaluation Report](https://www.w3.org/TR/PFE-evaluation/)
+simulated performance with two strategies: Range Request and Patch Subset.
+
+Performance was simulated on
+[**different speeds of network**](https://www.w3.org/TR/PFE-evaluation/#network)
+(from fast wired to 2G),
+for [**three classes of writing system**](https://www.w3.org/TR/PFE-evaluation/#langtype)
+(simple alphabetic, complex shaping, and large)
+and for two methods (Range Request and Patch Subset, see below).
+
+Both **size** (total bytes transferred, including overhead)
+and **network cost** (impact of latency on time to render)
+were considered.
+
+These initial approaches informed the current specification,
+but the WG ended up discarding one and substantially rewriting the other.
+
+### Range Request
+
+The Range Request method relied on the existing HTTP Range Request functionality
+and therefore could be used with **any HTTP server**.
+For best efficiency, the font needed to be 
+[re-ordered](https://w3c.github.io/IFT/RangeRequest.html#font-organization)
+before upload to the server.
+
+The client still needed to be updated to support this method.
+
+Work on Range Request was discontinued because:
+
+- Performance was insufficient in general
+- Only glyph outlines were subsetted
+- On slower networks, Range Request was [worse then no subsetting at all](https://www.w3.org/TR/PFE-evaluation/#conclusions-shaping)
+
+### Patch Subset
+
+The Patch Subset method required the server to 
+[**dynamically respond to a PatchRequest**](https://www.w3.org/TR/2023/WD-IFT-20230530/#handling-patch-request)
+by validating the request,
+computing a binary patch between the current, subsetted font on the client
+and the desired subset of the original font,
+and then sending the patch,
+which the client applied to produce a new, enlarged subset font.
+
+It therefore required new server capabilities,
+in addition to client changes.
+
+Work on Patch Subset was discontinued because, despite
+[median byte reductions of 90% for large fonts](https://www.w3.org/TR/PFE-evaluation/#analysis-cjk),
+
+- The dynamic subsetting severely impacted CDN **cache performance**
+- There was no subsetting of _design axes_, an unacceptable result with **variable fonts** becoming ubiquitous
+- Subsetting fonts with **complex interactions between glyphs** was challenging
+- Requiring an intelligent, dynamic server hindered widespread deployment
+- Very fine-grained subsetting [might be a privacy violation](https://www.w3.org/TR/2023/WD-IFT-20230530/#content-inference-from-character-set)

```

Comment:
Another draw back is that it required a custom protocol (and http header) to communicate with the dynamic backend.